test/test_match_ids.py

```diff
@@ -0,0 +1,44 @@
+import unittest
+import os
+import sys
+
+from normalize_license_text.normalize_class import NormalizeText
+from configuration.config import PACKAGE_PATH
+
+input_text = PACKAGE_PATH + "/test/data/test_all_ids.txt"
+directory = PACKAGE_PATH + "/data/templates/"
+
+
+class TestAllTexts(unittest.TestCase):
+    def test_main_script(self):
+        with open(input_text, 'r') as inputfile:
+            input_text_string = inputfile.read()
+            inputfile.close()
+            x = NormalizeText(input_text_string)
+            normalized_text_string = x.returnfinalstring_for_template()
+
+        for filename in os.scandir(directory):
+            file_name = str(filename.path)
+            file_name = file_name.replace(str(directory), '')
+            print(file_name)
+
+            try:
+                with open(filename.path, 'r') as input_file:
+                    input_template_file = input_file.read()
+                    input_file.close()
+                    object_normalization = NormalizeText(input_template_file)
+                    input_template_file = object_normalization.returnfinalstring_for_template()
+
+                    y = NormalizeTemplate(
+                        normalized_text_string, input_template_file)
+                    y.normalize_template()
+                    normalized_template_string = y.return_normalized_template()
+            except BaseException:
+                continue
+
+            if(CompareNormalizedFiles(normalized_template_string, normalized_text_string)):
+                print("The Text and the Template- " + file_name + " Match.")
```

Comment:
There is no assert in the test. You should assert that the final string printed by the function matches what we expect. Also, no need to normalize the license in the test itself. We need to test the code of `match_against_all_templates/match_all_ids.py` , here you have implemented the whole code itself.

Reply:
@rtgdk Actually I am having issues in implementing this test. How exactly would I test a command-line script? This test isn't returning anything actually. Its just printing the licenses which are matching. That is why the main thing which is to test is the Compare License Template Function which I have checked in this file `test_match_ids.py`. If I check just the match_all_ids.py, it will just assert whether input is present or not.

Reply:
Take an example file and try to match it against all the other licenses and assert only matched licenses returns CompareNormalizedFiles(normalized_template_string, normalized_text_string) as true. (try to keep test files where there is little differences between the license text and it returns true for multiple cases)